homeassistant/components/philips_js/binary_sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Philips TV binary sensors."""
+from __future__ import annotations
+
+from haphilipsjs import PhilipsTV
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PhilipsTVDataUpdateCoordinator
+from .const import DOMAIN
+
+RECORDINGS_LIST = "recordings"
+RECORDINGS_API_VERSION = 6
+
+
+class PhilipsTVBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """A entity description for Philips TV binary sensor."""
+
+    def __init__(self, recording_entry, recording_value, *args, **kwargs) -> None:
+        """Set up a binary sensor entity description and add additional attributes."""
+        super().__init__(*args, **kwargs)
+        self.recording_entry: str = recording_entry
+        self.recording_value: str = recording_value
+
+
+DESCRIPTIONS = (
+    PhilipsTVBinarySensorEntityDescription(
+        key="recording_ongoing",
+        has_entity_name=True,
+        translation_key="recording_ongoing",
+        icon="mdi:record-rec",
+        recording_entry="RecordingType",
+        recording_value="RECORDING_ONGOING",
+    ),
+    PhilipsTVBinarySensorEntityDescription(
+        key="recording_new",
+        has_entity_name=True,
+        translation_key="recording_new",
+        icon="mdi:new-box",
+        recording_entry="RecordingType",
+        recording_value="RECORDING_NEW",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the configuration entry."""
+    coordinator: PhilipsTVDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+
+    if (
+        coordinator.api.json_feature_supported(RECORDINGS_LIST, "List")
```

Comment:
Partial constants. The RECORDINGS_LIST is not really named reasonable. Just drop them i think. Just write out "recordings" here. Since they are only used once i see no need for the constant here.

Reply:
Dropped them.